main.go

```diff
@@ -71,7 +78,47 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	ctx := ctrl.SetupSignalHandler()
+
+	restConfig := ctrl.GetConfigOrDie()
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		setupLog.Error(err, "Error creating Kubernetes client")
+		os.Exit(1)
+	}
+
+	// InstaScale configuration
+	cfg := config.InstaScaleConfiguration{
+		MaxScaleoutAllowed: 3,
+	}
+
+	// Source InstaScale ConfigMap
+	if InstaScaleConfigMap, err := kubeClient.CoreV1().ConfigMaps(configsNamespace).Get(ctx, "instascale-config", metav1.GetOptions{}); err != nil {
+		setupLog.Error(err, "Error reading 'instascale-config' ConfigMap")
+		os.Exit(1)
+	} else if maxScaleoutAllowed := InstaScaleConfigMap.Data["maxScaleoutAllowed"]; maxScaleoutAllowed != "" {
+		if max, err := strconv.Atoi(maxScaleoutAllowed); err != nil {
+			setupLog.Error(err, "Error converting 'maxScaleoutAllowed' to integer")
+			os.Exit(1)
```

Comment:
Should instaScale completely error out if the configmap is invalid? Or should it continue on with a default of 3.

Reply:
I guess in a way -- this configmap is something that would be set up by the ODH/RHODS at a later date when the integration is tighter.